frontend/src/metabase/visualizations/lib/apply_tooltips.js

```diff
@@ -23,22 +23,26 @@ export function getClickHoverObject(
     settings,
   },
 ) {
-  let { cols } = series[0].data;
+  let { cols } = series[seriesIndex].data;
   const { card } = series[seriesIndex];
 
   const isMultiseries = series.length > 1;
   const isBreakoutMultiseries = isMultiseries && card._breakoutColumn;
   const isBar = classList.includes("bar");
   const isSingleSeriesBar = isBar && !isMultiseries;
 
-  // always format the second column as the series name?
   function getColumnDisplayName(col) {
+    const seriesSettings =
+      settings &&
+      settings.series_settings &&
+      settings.series_settings[col.name];
+    const title = seriesSettings && seriesSettings.title;
```

Comment:
This could be `getIn(settings, ["series_settings", col.name, "title"])` to save some lines.